mesonbuild/interpreter.py

```diff
@@ -47,6 +47,9 @@
                            'sources'},
 }
 
+known_langs = ['c', 'cpp', 'objc', 'objcpp', 'cuda', 'java', 'cs', 'vala', 'd',
+               'rust', 'fortran', 'swift']
```

Comment:
This seems off. It should be obtained automatically (somehow? introspecting compiler class names?) or, if that is not possible, this should be in the same place as all other language definitions, such as `compilers.py`.

Reply:
Oh yes, this already exists as `compilers.all_languages`

The list of languages is still in two places (there, and the mapping between language name and compiler detector function encoded by `compiler_from_language`), but let's not add a third :)